src/utils/ParseMachineState.ts

```diff
@@ -1,4 +1,7 @@
 export const ParseMachineState = ({ value, context }) => {
+    if(value !== "published") {
+        value = Object.keys(value)[0]
```

Comment:
Add a comment here to explain why you need to get the first key and why value is an Object rather than a string.

Additionaly, you should add a check wether value is an Object before extracting the key.

Reply:
The states assigned and reported has states compound, and when we want to save the machine value, we get "assigned: undraft" for example or "reported: draft". If we save the value in this way, when we try to persiste state, the machine don't recognized the initial value 'cause the value it's a object. So for these states i get only the key value to save in dataBase, and for others states i save normally

Reply:
I got it, please add a comment in the code for posteriority.